hydra/extra/callbacks.py

```diff
@@ -0,0 +1,32 @@
+from omegaconf import DictConfig
+
+
+class Callbacks:
+    """
+    Provides callbacks through different life cycle events of an Hydra application.
+    """
+
+    @abstractmethod
+    def setup(
+        self,
+        config: DictConfig,
+    ) -> None:
+        raise NotImplementedError()
+
+    def pre_run(self, cfg: DictConfig) -> None:
```

Comment:
We should document the diffrence between `run`, `sweep` and `job`

Reply:
Agreed!